runtime/drivers/duckdb/connectors.go

```diff
@@ -77,30 +81,38 @@ func (c *connection) ingestLocalFiles(ctx context.Context, env *connectors.Env,
 	// Not using query args since not quite sure about behaviour of injecting table names that way.
 	// Also, it's a source, so the caller can be trusted.
 
-	var from string
-	if conf.Format == ".csv" && conf.CSVDelimiter != "" {
-		from = fmt.Sprintf("read_csv_auto(['%s'], delim='%s')", path, conf.CSVDelimiter)
-	} else {
-		from, err = getSourceReader(localPaths)
-		if err != nil {
-			return err
-		}
+	from, err := getSourceReader(localPaths, conf)
+	if err != nil {
+		return err
 	}
 
 	qry := fmt.Sprintf("CREATE OR REPLACE TABLE %s AS (SELECT * FROM %s)", source.Name, from)
 
 	return c.Exec(ctx, &drivers.Statement{Query: qry, Priority: 1})
 }
 
-func getSourceReader(paths []string) (string, error) {
-	ext := fileutil.FullExt(paths[0])
-	if ext == "" {
+func getSourceReader(paths []string, config *connectors.Config) (string, error) {
+	format := config.Format
+	if format == "" {
+		format = fileutil.FullExt(paths[0])
+	}
```

Comment:
Are we sure `len(paths) > 0`? If not, maybe make `ConsumeAsFiles` guarantee to return an error if no data was returned

Reply:
Yes `ConsumeAsFiles` already returns error if no data is returned.